mantidproject/mantid
# Refactor ISIS ORSO tests

### Description of work

#### Summary of work

This PR refactors the `SaveISISReflectometryORSO` tests to address the items described in the linked issue. The unit tests now patch the workspace history to allow more specific and faster tests to be written for a wider variety of scenarios. System tests have been added to provide a thorough check of the output file. The system tests run the full ISIS Reflectometry reduction so that we can pick up any changes to the reduction that impact on the file output.



Fixes #36777. 




### To test:


All tests should pass. I have run a build from branch to check that the new system tests pass on all operating systems.

*This does not require release notes* because **it is only a change to tests, there are no user-facing changes.**



---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
